src/gfx.cpp

```diff
@@ -1445,20 +1438,78 @@ void RedrawScreenRect(int left, int top, int right, int bottom)
 	VideoDriver::GetInstance()->MakeDirty(left, top, right - left, bottom - top);
 }
 
+static std::vector<Rect> _dirty_viewport_occlusions;
+static const Viewport *_dirty_viewport;
```

Comment:
missing include for `Viewport` ?

Reply:
I missed a few renames when I backed out https://github.com/OpenTTD/OpenTTD/pull/8260. Fixed now.